pkg/virt-launcher/virtwrap/device/hostdevice/sriov/networkstatus-pcipool.go

```diff
@@ -0,0 +1,127 @@
+		return err
+	}
+	sriovNetworkToMultusInterfaceNameMap := loadNetworkNameToMultusInterfaceName(sriovIfaces, multusNetworks)
+
+	for networkName, multusInterfaceName := range sriovNetworkToMultusInterfaceNameMap {
+		networkStatusEntry := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
+		if networkStatusEntry.DeviceInfo == nil || networkStatusEntry.DeviceInfo.Pci == nil {
+			return errors.New(fmt.Sprintf("SRIOV interface %s has no corresponding device-info/pci-address in the appropriate network status annotations", networkName))
+		}
+
+		pciAddress := networkStatusEntry.DeviceInfo.Pci.PciAddress
+		if pciAddress == "" {
+			return errors.New(fmt.Sprintf("failed to associate pci-address to SRIOV interface %s", networkName))
+		}
+		p.networkNameToPciAddress[networkName] = pciAddress
+	}
+	return nil
+}
+
+func (p *PCIAddressPoolWithNetworkStatus) Pop(networkName string) (string, error) {
+	pciAddress, exists := p.networkNameToPciAddress[networkName]
+	if !exists {
+		return "", fmt.Errorf("resource for SR-IOV network %s does not exist", networkName)
+	}
+
+	delete(p.networkNameToPciAddress, networkName)
+	return pciAddress, nil
+}
+
+func loadNetworkNameToMultusInterfaceName(interfaceList []v1.Interface, networkList []v1.Network) map[string]string {
+	networkToMultusInterfaceNameMap := map[string]string{}
+	for i, network := range networkList {
+		// multusInterfaceName follows a predefined indexation of the non-default multus networks (See pkg/virt-controller/services/multus_annotations.go: generateMultusCNIAnnotation.
+		// In order to not break this convention, this for loop should not break/continue
+		multusInterfaceName := fmt.Sprintf("net%d", i+1)
+		associatedNetwork := getIfaceByNetworkName(interfaceList, network.Name)
+		if associatedNetwork != nil {
+			networkToMultusInterfaceNameMap[network.Name] = multusInterfaceName
+		}
+	}
+	return networkToMultusInterfaceNameMap
+}
+
+func loadMultusInterfaceNameToNetworkStatus(vmiAnnotations map[string]string) (map[string]networkv1.NetworkStatus, error) {
+	multusInterfaceNameToNetworkStatusMap := map[string]networkv1.NetworkStatus{}
+	networkStatuses, err := getNetworkStatusList(vmiAnnotations)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, networkStatus := range networkStatuses {
+		multusInterfaceNameToNetworkStatusMap[networkStatus.Interface] = networkStatus
+	}
+	return multusInterfaceNameToNetworkStatusMap, nil
+}
+
+func getIfaceByNetworkName(ifaces []v1.Interface, networkName string) *v1.Interface {
+	for i, iface := range ifaces {
+		if iface.Name == networkName {
+			return &ifaces[i]
+		}
+	}
+	return nil
+}
+
+func getNetworkStatusList(vmiAnnotations map[string]string) ([]networkv1.NetworkStatus, error) {
+	netStatusesJson, isSet := vmiAnnotations[networkv1.NetworkStatusAnnot]
+	if !isSet {
+		return nil, errors.New(fmt.Sprintf("%s annotation is not present on vmi annotations", networkv1.NetworkStatusAnnot))
+	}
+
+	var netStatuses []networkv1.NetworkStatus
```

Comment:
nit: Please consider this change so the returned value's name will match to the function's name:
```suggestion
	var networkStatusList []networkv1.NetworkStatus
```

Reply:
Done